lib/chef/handler/datadog.rb

```diff
@@ -118,6 +119,83 @@ def restore_env_proxies
         ENV['http_proxy'] = @env_http_proxy
         ENV['https_proxy'] = @env_https_proxy
       end
+
+      # create and configure all the Dogapi Clients to be used
+      #
+      # @return [Array] all Dogapi::Client to be used
+      def prepare_the_pack
+        dogs = []
+        endpoints.each do |url, api_key, app_key|
+          dogs.push(Dogapi::Client.new(
+                      api_key,
+                      app_key,
+                      nil,   # host
+                      nil,   # device
+                      true,  # silent
+                      nil,   # timeout
+                      url
+          ))
+        end
+        dogs
+      end
+
+      def convert_to_array(value_or_array)
+        return value_or_array if value_or_array.nil?
+        value_or_array.is_a?(Array) ? value_or_array : [value_or_array]
+      end
+
+      # return all endpoints as a list of triplets [url, api_key, application_key]
+      def endpoints
+        urls = convert_to_array @config[:url]
+        api_keys = convert_to_array @config[:api_key]
+        app_keys = convert_to_array @config[:application_key]
+
+        validate_extra_endpoints(urls, api_keys, app_keys)
+
+        endpoints = []
+        if urls.nil? || urls.length == 1
+          url = urls.nil? ? nil : urls[0]
+          [api_keys.length, app_keys.length].min.times do |index|
```

Comment:
It writes an error but doesn't fail, only an absence of `api_key` or `application_key` raises an error.

I wrote this so that it always tries to send something, even if the configuration is not completely correct.